File: backend/models/GroceryCategory.js
const mongoose = require('mongoose');

const groceryCategorySchema = new mongoose.Schema(
    {
        name: {
            type: String,
            required: [true, 'Category name is required'],
            unique: true,
            trim: true,
        },
        emoji: {
            type: String,
            required: [true, 'Emoji is required'],
        },
        description: {
            type: String,
            default: '',
        },
        image: {
            type: String,
            default: '',
        },
        sortOrder: {
            type: Number,
            default: 0,
        },
        isActive: {
            type: Boolean,
            default: true,
        },
    },
    {
        timestamps: true,
    }
);

// Indexes
groceryCategorySchema.index({ sortOrder: 1 });
groceryCategorySchema.index({ isActive: 1 });

const GroceryCategory = mongoose.model('GroceryCategory', groceryCategorySchema);

module.exports = GroceryCategory;

File: packages/grab_go_customer/lib/features/groceries/model/grocery_category.dart
class GroceryCategory {
  final String id;
  final String name;
  final String emoji;
  final String description;
  final String image;
  final int sortOrder;
  final bool isActive;

  GroceryCategory({
    required this.id,
    required this.name,
    required this.emoji,
    required this.description,
    required this.image,
    required this.sortOrder,
    required this.isActive,
  });

  factory GroceryCategory.fromJson(Map<String, dynamic> json) {
    return GroceryCategory(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      emoji: json['emoji'] ?? '',
      description: json['description'] ?? '',
      image: json['image'] ?? '',
      sortOrder: json['sortOrder'] ?? 0,
      isActive: json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'emoji': emoji,
      'description': description,
      'image': image,
      'sortOrder': sortOrder,
      'isActive': isActive,
    };
  }

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is GroceryCategory && other.id == id;
  }

  @override
  int get hashCode => id.hashCode;
}
